src/utils/typedData/index.ts

```diff
@@ -102,6 +108,18 @@ const encodeValue = (typedData: TypedData, type: string, data: unknown): [string
     return [type, getStructHash(typedData, type, data as Record<string, unknown>)];
   }
 
+  if (
+    Object.keys(typedData.types)
```

Comment:
Could `typedData.types` have felt as a key?
If `typedData.types` have felt as a key and `type` is felt, line 123 will never get executed AND we return `computeHashOnElements(structHashes)` instead of `computeHashOnElements(data as string[])`
Is this ok?